Added the google_trends example to community examples , edited the re… (#507)
fivetran/fivetran_connector_sdk: connectors/google_trends/README.md

Refer to the [Connector SDK Setup Guide](https://fivetran.com/docs/connectors/connector-sdk/setup-guide) to get started.

## Features

- Full refresh sync strategy – Fetches complete timeframes on every sync to track historical changes
- Multiple keyword comparison – Query up to 5 keywords per search group with relative interest scores
- Multi-region support – Fetch data for worldwide or specific country regions in a single sync
- Search grouping – Keywords in the same search group are normalized together for accurate comparison
- Incremental historical tracking – Append-only design with sync timestamps preserves data evolution
- Exponential backoff retry – Automatic retry with randomized delays (60-120s, 120-180s, 240-300s) handles rate limiting
- Partial data detection – Identifies incomplete data points marked as partial by Google Trends
- Flexible timeframe formats – Supports both relative ("today 12-m") and absolute date ranges ("2024-01-01 2026-02-03")

## Configuration file

The connector requires a `configuration.json` file with a `searches` array. Each search defines a group of keywords to compare, the regions to query, and the timeframe to analyze.

```json

> "[{"name":"ETL Tools Comparison","keywords":["Fivetran","Airflow","dbt"],"regions":[{"name":"Worldwide","code":""},{"name":"United States","code":"US"},{"name":"United Kingdom","code":"GB"}],"timeframe":"2024-01-01 today"}]"

Reviewer: Does it require formatting?

```

Configuration parameters:
- `searches` – Array of search configurations (required)
  - `name` – Human-readable name for the search group
  - `keywords` – Array of 1-5 keywords to compare (required, Google Trends limit)
  - `regions` – Array of region objects (required)
    - `name` – Display name for the region
    - `code` – ISO country code (empty string "" for worldwide)
  - `timeframe` – Time period in Google Trends format (required)
    - Relative format: "today 12-m" (last 12 months), "today 3-m" (last 3 months)
    - Absolute format: "2024-01-01 2026-02-03" (specific date range)

>     - Note: "today" keyword in absolute ranges is automatically converted to current date

Reviewer: ```suggestion
    Note: "today" keyword in absolute ranges is automatically converted to current date
```

Note: Ensure that the `configuration.json` file is not checked into version control to protect sensitive information.

## Requirements file

The `requirements.txt` file specifies the Python libraries required by the connector. This connector requires the PyTrends library for accessing Google Trends data and Pandas for data processing.

```
# Google Trends API client library
pytrends==4.9.2

# Pandas for data processing (required by PyTrends)
pandas==3.0.0
```

Note: The `fivetran_connector_sdk:latest` and `requests:latest` packages are pre-installed in the Fivetran environment. To avoid dependency conflicts, do not declare them in your `requirements.txt`.

## Authentication

> This connector uses the public Google Trends API, which does not require authentication credentials. The connector accesses publicly available search interest data without needing API keys, OAuth tokens, or user credentials.

Reviewer: ```suggestion
This connector uses the public Google Trends API, which does not require authentication credentials. The connector accesses publicly available search interest data without the need for API keys, OAuth tokens, or user credentials.
```

   - `search_id` – Groups keywords from the same search
   - `keyword` – The search term
   - `date` – ISO-formatted timestamp (UTC)
   - `region_name` – Display name of the region
   - `region_code` – ISO country code or "WORLDWIDE"
   - `sync_timestamp` – ISO timestamp of when this sync ran
   - `interest` – Interest score (0-100 scale)
   - `timeframe` – Original timeframe query string
   - `is_partial` – Boolean flag for incomplete data points

   - Data upserting – Each record is upserted to the `google_trends` table. The combination of `search_id`, `keyword`, `date`, `region_code`, and `sync_timestamp` forms the primary key, ensuring each sync creates new records for historical tracking. Refer to `def process_region()`.

## Error handling

The connector implements robust error handling to ensure reliable data synchronization:

- **Retry logic with exponential backoff**
  - Automatically retries failed API requests up to 5 times
  - Uses exponential backoff with random jitter to avoid thundering herd issues
  - See `def fetch_region_data_with_retry()` and `def calculate_retry_delay()`

> - **Regional failure tolerance**

Reviewer: ```suggestion
- Regional failure tolerance
```

  - Continues syncing other regions if one region fails
  - Sync fails only if all regions fail or no data is returned
  - See `def update()` in `connector.py`

> - **Detailed error logging**

Reviewer: ```suggestion
- Detailed error logging
```
